test/BiconomyForwarderTest.js

```diff
@@ -0,0 +1,363 @@
+        {name:'nonce',type:'uint256'},
+        {name:'dataHash',type:'bytes32'}
+    ];
+
+    before(async function(){
+
+        accounts = await ethers.getSigners();
+
+        const Forwarder = await ethers.getContractFactory("BiconomyForwarder");
+        forwarder = await Forwarder.deploy();
+        await forwarder.deployed();
+
+        const TestRecipient = await ethers.getContractFactory("TestRecipient");
+        testRecipient = await TestRecipient.deploy(forwarder.address);
+        await testRecipient.deployed();
+
+        domainData = {
+            name : "TestRecipient",
+            version : "1",
+            chainId : 42,
+            verifyingContract : forwarder.address
+          };
+
+        await forwarder.registerDomainSeparator("TestRecipient","1");
+        domainSeparator = ethers.utils.keccak256((ethers.utils.defaultAbiCoder).
+                          encode(['bytes32','bytes32','bytes32','uint256','address'],
+                                 [ethers.utils.id("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"),
+                                 ethers.utils.id(domainData.name),ethers.utils.id(domainData.version),
+                                 domainData.chainId,domainData.verifyingContract]));
+        
+        const TestnetDai = await ethers.getContractFactory("TestnetDAI");
+        testnetDai = await TestnetDai.deploy();
+        await testnetDai.deployed();
+
+        //deploy fee multiplier with a factor of 1.5x
+        const MockFeeMultiplier = await ethers.getContractFactory("MockFeeMultiplier");
+        mockFeeMultiplier = await MockFeeMultiplier.deploy();
+        await mockFeeMultiplier.deployed();
+        await mockFeeMultiplier.setFeeMultiplier(15000);
+
+        //deploy and fill up faucet
+        const Faucet = await ethers.getContractFactory("mockFaucet");
+        faucet = await Faucet.deploy();
+        await faucet.deployed();
+        await accounts[0].sendTransaction({value:ethers.utils.parseEther("100"),to:faucet.address});
+
+
+    });
+
+    describe("personal sign", function(){
+        it("executes call successfully", async function(){
+            const req = await testRecipient.populateTransaction.doCall(await accounts[1].getAddress());
+            req.from = await accounts[1].getAddress();
+            req.nonce = (await forwarder.getNonce(req.from)).toNumber();
+            req.gas = (req.gasLimit).toNumber();
+            req.price = 0;
+            req.msgValue = 0;
+            delete req.gasPrice;
+            delete req.gasLimit;
+            delete req.chainId;
+            req.token = testnetDai.address;
+            req.feeReceiver = await accounts[0].getAddress();
+            req.feeMultiplierManager = mockFeeMultiplier.address;
+            const hashToSign = abi.soliditySHA3(['address','address','address','address','address',
+                                                 'uint256','uint256','uint256','uint256','bytes32'],
```

Comment:
last type should be bytes instead of bytes32

Reply:
I made this decision due to encoding issues I was having. Will see if I can make this change